dlp/custom_infotype.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Custom infoType snippets.
+
+This file contains sample code that uses the Data Loss Prevention API to create
+custom infoType detectors to refine scan results.
+"""
+
+
+# [START dlp_omit_name_if_also_email]
+def omit_name_if_also_email(
+    project,
+    content_string,
+):
+    """Marches PERSON_NAME and EMAIL_ADDRESS, but not both.
+
+    Uses the Data Loss Prevention API omit matches on PERSON_NAME if the
+    EMAIL_ADDRESS detector also matches.
+    Args:
+        project: The Google Cloud project id to use as a parent resource.
+        content_string: The string to inspect.
+
+    Returns:
+        None; the response from the API is printed to the terminal.
+    """
+
+    # Import the client library.
+    import google.cloud.dlp
+
+    # Instantiate a client.
+    dlp = google.cloud.dlp_v2.DlpServiceClient()
+
+    # Construct a list of infoTypes for DLP to locate in `content_string`
+    info_types_to_locate = [{"name": "PERSON_NAME"}, {"name": "EMAIL_ADDRESS"}]
```

Comment:
nit: Can you add a link to the docs where infoTypes are defined so users can find these magic strings?

Reply:
Sure thing!